### PR TITLE
tests: Ignore NotImplemented error in listenBucketNotification

### PR DESCRIPTION
**Pull request by vadmeste**
Functional tests were just infinitely waiting for a put object
notification event, this commit will just ignore NotImplemented
error to support gateways which are not supporting bucket
notifications.

### CODE REVIEW

**Comment by vadmeste** [comment]
@harshavardhana your comment is addressed, the test ignores not implemented error now.

(by the way, listen bucket notification is supported in gateway NAS)

**Comment by harshavardhana** [comment]
I don't know which release you are using it should be disabled as per https://github.com/minio/minio/pull/6920

**Comment by poornas** [comment]
it is defined incorrectly - should have been
```
func (n *nasObjects) IsListenBucketSupported() bool {
	return false
}
```

**Comment by harshavardhana** [comment]
> it is defined incorrectly - should have been
> 
> ```
> func (n *nasObjects) IsListenBucketSupported() bool {
> 	return false
> }
> ```

Please fix it @poornas it is not expected to support ListenBucket with NAS gateway.

File: functional/FunctionalTest.java [hunk 1]
```diff
@@ -2856,19 +2861,43 @@ public void run() {
...
 
+      // Fail for any exception but ignore NotImplemented error
+      if (bl.lastException != null) {
+        ErrorResponse errorResponse = null;
+        if (bl.lastException instanceof ErrorResponseException) {
+          ErrorResponseException exp = (ErrorResponseException) bl.lastException;
+          errorResponse = exp.errorResponse();
+        }
+        if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
+          mintIgnoredLog("listenBucketNotification(String bucketName)", null, startTime);
```
Comment (kannappanr) [inline functional/FunctionalTest.java]: what about cleanup here?

File: functional/FunctionalTest.java [hunk 2]
```diff
@@ -2856,19 +2861,43 @@ public void run() {
...
+          return;
+        } else {
+          mintFailedLog("listenBucketNotification(String bucketName)", null, startTime,
+              null, bl.lastException.toString() + " >>> " + Arrays.toString(bl.lastException.getStackTrace()));
```
Comment (kannappanr) [inline functional/FunctionalTest.java]: Here also, created bucket and object will not be removed.

File: functional/FunctionalTest.java [hunk 3]
```diff
@@ -2856,19 +2861,43 @@ public void run() {
...
+      for (int i = 0; i < 20; i++) {
         client.putObject(bucketName, "prefix-random-suffix", file1kb);
-        Thread.sleep(300);
+        Thread.sleep(500);
+        if (bl.eventsReceived > 0 || bl.lastException != null) {
+          break;
+        }
       }
 
+      // Fail for any exception but ignore NotImplemented error
```
Comment (kannappanr) [inline functional/FunctionalTest.java]: Maybe the cleanup code of removing the bucket and object should be moved to this part before you start checking for errors.

Comment (vadmeste) [reply functional/FunctionalTest.java]: > Maybe the cleanup code of removing the bucket and object should be moved here before you start checking for errors.

Nice catch! fixed

**Comment by poornas** [comment]
@vadmeste, this needs rebase and build failures checked.

**Pull request merged**